src/Handlers.cpp.Rt

```diff
@@ -897,49 +897,105 @@ class cbFailcheck: public Callback {
 		reg.dx = solver->region.dx;
 		reg.dy = solver->region.dy;
 		reg.dz = solver->region.dz;
-               	pugi::xml_attribute attr = node.attribute("dx");
-                if (attr) {
-                       	reg.dx = solver->units.alt(attr.value());
-                }
-                attr = node.attribute("dy");
-                if (attr) {
-                        reg.dy = solver->units.alt(attr.value());
-                }
-                attr = node.attribute("dz");
-                if (attr) {
-                       	reg.dz = solver->units.alt(attr.value());
-                }
+	
+
+
+        pugi::xml_attribute attr = node.attribute("dx");
+        if (attr) {
+            reg.dx = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("dy");
+        if (attr) {
+            reg.dy = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("dz");
+        if (attr) {
+            reg.dz = solver->units.alt(attr.value());
+        }
+
+
+        attr = node.attribute("nx");
+        if (attr) {
+            reg.nx = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("ny");
+        if (attr) {
+            reg.ny = solver->units.alt(attr.value());
+        }
+        attr = node.attribute("nz");
+        if (attr) {
+            reg.nz = solver->units.alt(attr.value());
+        }
+
 		rkept = 1;
 		return 0;
 	}
 	int DoIt() {
 		Callback::DoIt();
-        	int ret = 0;
-		int cond,fin;
+       	int ret = 0;
+		int cond,fin,go;
 		fin = false;
 		cond = false;
+        go = false;
+
+        pugi::xml_attribute comp = node.attribute("check_quantities");
+        std::vector<std::string> components;
+        std::vector<real_t*> buffers;
+        char *str = strdup(comp.value());  // We own str's memory now.
+        char *token;
+        int n_components = 0;
+        while ((token = strsep(&str, ","))) {
+            if (strlen(token) > 0 ){
+                components.push_back(std::string(token));
+                n_components++;
+            }
+        }
+        buffers.resize(n_components);
+        free(str);      
+
+
 		<?R for (q in rows(Quantities)) { ifdef(q$adjoint); ?>
+        
 		if (!cond) {
-			<?%s q$type ?> tmp;
-			solver->lattice->Get<?%s q$name ?>(reg,&tmp);
-			<?R if (q$vector) { ?>
-	                cond = ((std::isnan(tmp.x)) || (std::isnan(tmp.y)) || (std::isnan(tmp.z)));  
-			<?R } else { ?>  
-			cond = (std::isnan(tmp));
-			<?R } ?>
+            go = false;
+            for (int n = 0; n < n_components; n++) {
+                if ( strcmp(components[n].c_str(), "<?%s q$name ?>" ) == 0 ||  strcmp(components[n].c_str(), "ALL" ) == 0  ) {
+                    go = true;
+                }  
+            }
+            if (go || n_components == 0 ) {
+                notice("... checking <?%s q$name ?>");
```

Comment:
But in general, I don't think this kind of output in necessary. We should output only if something goes wrong. Maybe in the end of loop body, at line 981 for instance: 
```c
if (cond) notice("NaN value occured at <?%s q$name?>...");
```
will be better